### PR TITLE
CoW fortune

### LINKED ISSUE #1959: Cow Fortune
https://www.notion.so/cownation/CoW-fortune-say-hardcoded-3392c714f7434038bdc07f4a966ef8d0

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #1959

Specification: https://www.notion.so/cownation/FE-CoW-fortune-say-hardcoded-3392c714f7434038bdc07f4a966ef8d0

image

image

Config: https://github.com/cowprotocol/cowswap/blob/configuration/config/fortunes.json

### CODE REVIEW

File: src/cow-react/common/containers/FeatureGuard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useFeatureFlags } from '@cow/common/hooks/useFeatureFlags'
```
Comment (anxolin) [inline src/cow-react/common/containers/FeatureGuard/index.tsx:1]: Cool you did this FeatureGuard

It would be better to do it as an independent PR (also mention in the PR description)

File: src/cow-react/modules/application/containers/WithLDProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { LAUNCH_DARKLY_CLIENT_KEY } from '@cow/constants/launchDarkly'
+import { withLDProvider } from 'launchdarkly-react-client-sdk'
+import { PropsWithChildren } from 'react'
+
+function InnerWithLDProvider({ children }: PropsWithChildren) {
+  return <>{children}</>
+}
+export const WithLDProvider = withLDProvider<PropsWithChildren & JSX.IntrinsicAttributes>({
```
Comment (anxolin) [inline src/cow-react/modules/application/containers/WithLDProvider.tsx:8]: This is not technically part of cowfortune. 

Also, I wouldn't abbreviate to LD, i had no idea first when I saw it.

File: src/cow-react/modules/application/containers/App/index.tsx [hunk 3]
```diff
@@ -44,10 +42,3 @@ function WrappedApp() {
     </ErrorBoundary>
   )
 }
-
-export const App = withLDProvider({
-  clientSideID: LAUNCH_DARKLY_CLIENT_KEY,
-  options: {
-    bootstrap: 'localStorage',
-  },
-})(WrappedApp)
```
Comment (anxolin) [inline src/cow-react/modules/application/containers/App/index.tsx:53]: thanks for refactoring, looks better

**Review by anxolin — APPROVED** [review]
Approved! Only nitpick, i would prefer to separate the enhancement of Feature Flags from CoW Fortune, other than that, great (specially since I saw the re-iteration PR already)

**Review by anxolin — APPROVED** [review]

File: src/cow-react/common/containers/FeatureGuard/index.tsx (hunk 1 above)
Comment (tukantje) [reply src/cow-react/common/containers/FeatureGuard/index.tsx:1]: Will do that next time. It was needed to create a component for something like this to make the feature flag possible though.
